localstack/services/events/provider.py

```diff
@@ -573,8 +573,11 @@ def filter_event(event_pattern_filter: dict[str, Any], event: dict[str, Any]):
             elif isinstance(value, (str, dict)):
                 try:
                     value = json.loads(value) if isinstance(value, str) else value
-                    if isinstance(value, dict) and not filter_event(value, event_value):
-                        return False
+                    if isinstance(event_value, list):
```

Comment:
nice addition of being able to handle lists of operators following the AND logic 👍 👏